### PR TITLE
Deprecate prepare_for_use method in Stream class.

### LINKED ISSUE #204: [DEPRECATION] Deprecate prepare_for_use method in the Stream class
The `prepare_for_use` method adds an additional step during a stream initialization without providing additional functionality. Additionally, it has led to errors such as in #177 

### Current implementation:
```
stream = SomeStream()   # initialization but not ready to use
stream.preapare_for_use()
```

### Proposed change:
```
stream = SomeStream()   # initialization and ready to use
```

The proposal is to turn the existing `prepare_to_use` method into a private method to be called during initialization. The public method will become a `stub` with no impact on the stream. The rationale for this is:

1. The impact on current implementation is minimal.
2. The change is hidden from the user.
3. We will keep the public method for at least 2 minor versions as a `stub` method that will raise a `FutureWarning`.
4. Existing code will only require removing the deprecated `prepare_for_use`.

### PR DESCRIPTION
**Pull request by jacobmontiel**
Changes proposed in this pull request:

- Stream objects are ready to use after instantiation
- Streamline initialization of instances of the Stream class
- Streamlined stream restart methods
- Remove redundant code
- Update documentation to reflect this change
- Make some methods private in the stream generators.
- Add temporarily a FutureWarning to let the user know that prepare_for_use will be removed
- Update tests cases
- Fixes #204 and #177

**Note:** This is change does NOT affect the functionality of related methods, instances of the `Stream` class.

---

Checklist

- [x] Code complies with PEP-8 and is consistent with the framework.
- [x] Code is properly documented.
- [x] Tests are included for new functionality or updated accordingly.
- [x] Travis CI build passes with no errors.
- [x] Test Coverage is maintained (threshold is -0.2%).
- [x] Files changed (update, add, delete) are in the PR's scope (no extra files are included).

### CODE REVIEW

File: src/skmultiflow/data/base_stream.py [hunk 1]
```diff
@@ -236,7 +237,10 @@ def is_restartable(self):
 
     def restart(self):
         """  Restart the stream. """
-        self.prepare_for_use()
+        self.current_sample_x = None
+        self.current_sample_y = None
+        self.sample_idx = 0
+        self._prepare_for_use()
```
Comment (smastelini) [inline src/skmultiflow/data/base_stream.py]: Would not this call raise a `NotImplementedError`?

Comment (jacobmontiel) [reply src/skmultiflow/data/base_stream.py]: the `Stream` class is an abstract class so it should not be used as-is. Additionally, `_prepare_for_use` is an `@abstractmethod` so it must be implemented by all extensions of the `Stream` class.

Comment (smastelini) [reply src/skmultiflow/data/base_stream.py]: You are right. Sorry, I got confused. The change is correct!

**Comment by smastelini** [comment]
The changes are correct! Thanks for looking into that @jacobmontiel.

**Pull request merged**